horust/src/horust/formats/service.rs

```diff
@@ -624,6 +631,74 @@ impl From<TerminationSignal> for Signal {
     }
 }
 
+#[derive(Serialize, Clone, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct ResourceLimit {
+    #[serde(default)]
+    /// The CPU time that the process can use, non-positive value means no limit
+    pub(crate) cpu: f64,
```

Comment:
the default here I think it's going to be 0. Serde(default) will use the default implementation for the type, which for f64 is [0](https://doc.rust-lang.org/std/primitive.f64.html#impl-Default-for-f64). You will need to either use an Option and use -1 when unwrapping, or use a [function to return a default value](https://serde.rs/attr-default.html).

Reply:
Will change it to Option